lib/chef/util/dsc/lcm_output_parser.rb

```diff
@@ -53,7 +53,62 @@ module Parser
           #     )
           #   ]
           #
-          def self.parse(lcm_output)
+          # Example for TestDSCParser:
+          #   parse <<-EOF
+          #   InDesiredState            : False
+          #   ResourcesInDesiredState   :
+          #   ResourcesNotInDesiredState: {[Environment]texteditor}
+          #   ReturnValue               : 0
+          #   PSComputerName            : .
+          #   EOF
+          #
+          #   would return
+          #
+          #   [
+          #     Chef::Util::DSC::ResourceInfo.new(
+          #       '{[Environment]texteditor}',
+          #       true,
+          #       [
+          #       ]
+          #     )
+          #   ]
+          #
+
+          def self.parse(lcm_output, test_dsc_configuration)
```

Comment:
I'm not really sure what the purpose is of keeping parse around. Since we've broken into 2 functions, the caller should pick the right one.
Or, if we want to preserve some sort of backwards compatibility, `test_dsc_configuration` should default to false